src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs

```diff
@@ -76,30 +82,108 @@ protected virtual byte[] HashData(Stream data, HashAlgorithmName hashAlgorithm)
         public byte[] SignData(byte[] data, HashAlgorithmName hashAlgorithm)
         {
             if (data == null)
-            {
                 throw new ArgumentNullException(nameof(data));
-            }
-            return SignData(data, 0, data.Length, hashAlgorithm);
+            if (string.IsNullOrEmpty(hashAlgorithm.Name))
+                throw HashAlgorithmNameNullOrEmpty();
+
+            return SignDataCore(data, hashAlgorithm, DSASignatureFormat.IeeeP1363FixedFieldConcatenation);
```

Comment:
This changes the virtual dispatch order (used to be `SignData(byte[], HashAlgorithmName)` => `virtual SignData(byte[], int, int, HashAlgorithmName)` => `virtual HashData(byte[], int, int)` + `abstract CreateSignature(byte[])` now it's `SignData(byte[], HashAlgorithmName)` => `virtual SignDataCore(ReadOnlySpan<byte>, HashAlgorithmName, DSASignatureFormat)` => `virtual TrySignData(ReadOnlySpan<byte>, Span<byte>, HashAlgorithmName, DSASignatureFormat, out int)` => `virtual TryHashData(...)` (=> `virtual HashData(...)`) + `virtual TryCreateSignature(...)` (=> `abstract CreateSignature(byte[])`. -- The important part of the change is we no longer ever call `virtual SignData(byte[], int, int, HashAlgorithmName)`.

For .NET Framework we'd say no to a change like that.  I'm not sure how we feel in .NET Core.